src/app/(routes)/shopping-mall/components/filter-bar/FilterBar.tsx

```diff
@@ -0,0 +1,75 @@
+"use client";
+
+import React, { FC, useCallback } from "react";
+import { Bar, FilterButton, Filters, Span, Total } from "./FilterBar.styled";
+import { useRouter, useSearchParams } from "next/navigation";
+import { categoryEmojis } from "../../utils/get-emojis";
+import { Categories } from "../../context/category-context";
+
+type Props = {
+  activeFilter: string;
+  categories: Categories;
+  totalOfItems: number;
+};
+
+const FilterBar: FC<Props> = ({ activeFilter, categories, totalOfItems }) => {
+  const router = useRouter();
+  const searchParams = useSearchParams()!;
+
+  const createQueryString = useCallback(
+    (name: string, value: string) => {
+      const params = new URLSearchParams(searchParams);
+      params.set(name, value);
+
+      return params.toString();
+    },
+    [searchParams]
+  );
+
+  return (
+    <Bar>
+      <Filters>
+        <FilterButton
+          isActive={activeFilter === "all"}
+          type="button"
+          onClick={() => {
+            router.push(`/shopping-mall`);
+          }}
+        >
+          ♾️
+        </FilterButton>
+
+        {Object.keys(categories).map((category) => (
+          <FilterButton
+            isActive={activeFilter === category}
+            key={category}
+            type="button"
+            onClick={() => {
+              router.push(
+                `/shopping-mall?${createQueryString("category", category)}`
+              );
+            }}
+          >
+            {categoryEmojis.get(category as keyof Categories)
+              ? categoryEmojis.get(category as keyof Categories)
+              : category}
+          </FilterButton>
+        ))}
```

Comment:
When I see type castings such as `category as keyof Categories`, I personally consider them a "code smell" for non-optimal design, but I understand that sometimes it might be hard to avoid it. So at least you can optimize it a little bit:
```suggestion
        {(Object.keys(categories) as Array<keyof Categories>).map((category) => (
          <FilterButton
            isActive={activeFilter === category}
            key={category}
            type="button"
            onClick={() => {
              router.push(
                `/shopping-mall?${createQueryString("category", category)}`
              );
            }}
          >
            {categoryEmojis.get(category)
              ? categoryEmojis.get(category)
              : category}
          </FilterButton>
        ))}
```

Reply:
Fixed, also optimized the filter type inference. 💪